builders/nodejs.go

```diff
@@ -161,7 +161,43 @@ func (builder *NodeJSBuilder) IsModule(target string) (bool, error) {
 	return false, errors.New("IsModule is not implemented for NodeJSBuilder")
 }
 
-// DiscoverModules is not implemented
+// DiscoverModules builds ModuleConfigs for any package.jsons that are not contained in a node_modules dir
 func (builder *NodeJSBuilder) DiscoverModules(dir string) ([]config.ModuleConfig, error) {
-	return []config.ModuleConfig{}, errors.New("DiscoverModules is not implemented for NodeJSBuilder")
+	moduleConfigs := []config.ModuleConfig{}
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
```

Comment:
Why not use glob?

Reply:
Glob is really bad at excluding stuff, need to use walk.

Reply:
Glob and then filter the output. It'll be faster. Standard library Glob uses a syscall instead of walking to every file.

Reply:
Actually, it looks like you'll need a doublestar glob and I'm not sure what that performance profile looks like. We can roll with this for now and change it if it's noticeably bad.